config/account.go

```diff
@@ -0,0 +1,129 @@
+package config
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type AccountsFile struct {
+    ActiveToken string
+    Accounts []*Account
+}
+
+// Account has a name and a token. The name is just for the users recognition.
+// The token is the actual token used to authenticate against the discord API.
+type Account struct {
+	Name  string
+	Token string
+}
+
+var LoadedAccountsFile *AccountsFile
+var cachedAccountsFile string
+
+// SetAccountsFile sets the accounts file path cache to the
+// entered value
+func SetAccountsFile(accountsFilePath string) error {
+	// get parent directory of accounts file
+	parent := filepath.Dir(accountsFilePath)
+	err := ensureDirectory(parent)
+	if err == nil {
+		cachedAccountsFile = accountsFilePath
+	} else {
+		absolute, err := getAbsolutePath(parent)
+		if err != nil {
+			return err
+		}
+		err = ensureDirectory(absolute)
+		if err == nil {
+			cachedAccountsFile = accountsFilePath
+		}
+	}
+	return err
+}
+
+// GetTokenFile retrieves the config file path from cache
+// or sets it to the default config file location
+func GetAccountsFile() (string, error) {
```

Comment:
The following error occurs if no accounts file parameter is specified.

```
/home/ges/.config/cordless/accounts.jsonpanic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5634b7b8baee]

goroutine 9 [running]:
github.com/Bios-Marcel/cordless/app.Run.func2(0xc000279b80, 0x0, 0xc0001ca510, 0xc00002c0e0, 0x5634b84e1b80, 0xc00002c070)
	/home/ges/doc/src/github.com/kruhlmann/cordless/app/app.go:72 +0x4e
created by github.com/Bios-Marcel/cordless/app.Run
	/home/ges/doc/src/github.com/kruhlmann/cordless/app/app.go:66 +0x2c6
```

Reply:
This happens because the `getDefaultConfigDirectory` returns the full path of the config file in

https://github.com/Bios-Marcel/cordless/blob/f7ad359ef872f600cab92f2e231ad009970059fd/config/os.go#L24

Reply:
Fixed in b30fd34 and bb9319c